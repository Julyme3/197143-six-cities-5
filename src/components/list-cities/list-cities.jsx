import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {setSelectedCityAction} from "../../store/actions";
import {CitiesType} from "../../types";
import CitiesItem from "../cities-item/cities-item";
import {Cities} from "../../const";

class ListCities extends PureComponent {
  constructor(props) {
    super(props);

    this.handleCityClick = this.handleCityClick.bind(this);
  }

  componentWillUnmount() {
    const {setSelectedCity} = this.props;
    setSelectedCity(Cities[0]);
  }

  handleCityClick(selectedCity) {
    const {setSelectedCity} = this.props;
    setSelectedCity(selectedCity);
  }

  render() {
    const {cities, activeCity} = this.props;
    return (
      <section className="locations container">
        <ul className="locations__list tabs__list">
          {cities.length && cities.map((city) => {
            return (
              <CitiesItem
                key={city}
                city={city}
                activeCity={activeCity}
                onClickHandler={this.handleCityClick}
              />
            );
          }) || null}
        </ul>
      </section>
    );
  }
}

ListCities.propTypes = {
  cities: CitiesType,
  activeCity: PropTypes.string.isRequired,
  setSelectedCity: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  activeCity: state.PROCESS.activeCity,
});

const mapDispatchToProps = (dispatch) => ({
  setSelectedCity(selectedCity) {
    dispatch(setSelectedCityAction(selectedCity));
  },
});

export {ListCities};

export default connect(mapStateToProps, mapDispatchToProps)(ListCities);
