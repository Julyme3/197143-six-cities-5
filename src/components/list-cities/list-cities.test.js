import React from "react";
import renderer from "react-test-renderer";
import {ListCities} from "./list-cities";

export const Cities = [
  `Amsterdam`,
  `Cologne`,
  `Brussels`,
  `Paris`,
  `Hamburg`,
  `Dusseldorf`,
];

jest.mock(`../cities-item/cities-item`, () => `CitiesItem`);

test(`ListCities render`, () => {
  const tree = renderer
    .create(
        <ListCities
          cities={Cities}
          activeCity={Cities[0]}
          setSelectedCity={()=>{}}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
