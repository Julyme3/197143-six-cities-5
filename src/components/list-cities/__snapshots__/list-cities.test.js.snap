// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListCities render 1`] = `
<section
  className="locations container"
>
  <ul
    className="locations__list tabs__list"
  >
    <CitiesItem
      activeCity="Amsterdam"
      city="Amsterdam"
      onClickHandler={[Function]}
    />
    <CitiesItem
      activeCity="Amsterdam"
      city="Cologne"
      onClickHandler={[Function]}
    />
    <CitiesItem
      activeCity="Amsterdam"
      city="Brussels"
      onClickHandler={[Function]}
    />
    <CitiesItem
      activeCity="Amsterdam"
      city="Paris"
      onClickHandler={[Function]}
    />
    <CitiesItem
      activeCity="Amsterdam"
      city="Hamburg"
      onClickHandler={[Function]}
    />
    <CitiesItem
      activeCity="Amsterdam"
      city="Dusseldorf"
      onClickHandler={[Function]}
    />
  </ul>
</section>
`;
